### PR TITLE
Add two sample dapps: HTTPS get requests

### PR DESCRIPTION
**Pull request by dprats**
# Purpose

The purpose of this is to create a Motoko and Rust dapp that gives developers a minimal dapp that uses the IC's HTTPS outcalls feature to make a GET request.

This demo goes in hand with the [developer documentation on HTTPS outcalls](https://internetcomputer.org/docs/current/developer-docs/integrations/https-outcalls/https-outcalls-get).

### CODE REVIEW

**Review by jessiemongeon1 — APPROVED** [review]

File: rust/https_outcalls_GET_request/src/hello_http_rust_backend/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+async fn get_icp_usd_exchange() -> String {
+
+  //2. SETUP ARGUMENTS FOR HTTP GET request
+
+    // 2.1 Setup the URL and its query parameters
+    type Timestamp = u64;
+    let start_timestamp : Timestamp = 1682978460; //May 1, 2023 22:01:00 GMT
+    let seconds_of_time : u64 = 60; //we start with 60 seconds
+    let host = "api.pro.coinbase.com";
+    let url = format!("https://{}/products/ICP-USD/candles?start={}&end={}&granularity={}", host, start_timestamp.to_string(), start_timestamp.to_string(),seconds_of_time.to_string());
```
Comment (adamspofford-dfinity) [inline rust/https_outcalls_GET_request/src/hello_http_rust_backend/src/lib.rs:19]:
```suggestion
    let url = format!("https://{host}/products/ICP-USD/candles?start={start_timestamp}&end={start_timestamp}&granularity={seconds_of_time}");
```

File: rust/https_outcalls_GET_request/src/hello_http_rust_backend/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,89 @@
+//1. DECLARE IC MANAGEMENT CANISTER
+//This also includes methods and types needed
```
Comment (adamspofford-dfinity) [inline rust/https_outcalls_GET_request/src/hello_http_rust_backend/src/lib.rs:2]:
```suggestion
//1. IMPORT IC MANAGEMENT CANISTER
//This includes all methods and types needed
```

File: rust/https_outcalls_GET_request/src/hello_http_rust_backend/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+//Update method using the HTTPS outcalls feature
+#[ic_cdk::update]
+async fn get_icp_usd_exchange() -> String {
+
+  //2. SETUP ARGUMENTS FOR HTTP GET request
+
+    // 2.1 Setup the URL and its query parameters
+    type Timestamp = u64;
+    let start_timestamp : Timestamp = 1682978460; //May 1, 2023 22:01:00 GMT
```
Comment (adamspofford-dfinity) [inline rust/https_outcalls_GET_request/src/hello_http_rust_backend/src/lib.rs:16]: This code in general should be run through `cargo fmt`.

File: rust/https_outcalls_GET_request/src/hello_http_rust_backend/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    let seconds_of_time : u64 = 60; //we start with 60 seconds
+    let host = "api.pro.coinbase.com";
+    let url = format!("https://{}/products/ICP-USD/candles?start={}&end={}&granularity={}", host, start_timestamp.to_string(), start_timestamp.to_string(),seconds_of_time.to_string());
+
+    // 2.2 prepare headers for the system http_request call
+    //Note that `HttpHeader` is declared in line 4
+    let request_headers = vec![
+        HttpHeader { name: "Host".to_string(), value: format!("{}:443",host) },
```
Comment (adamspofford-dfinity) [inline rust/https_outcalls_GET_request/src/hello_http_rust_backend/src/lib.rs:24]:
```suggestion
        HttpHeader { name: "Host".to_string(), value: format!("{host}:443") }, 
```

File: rust/https_outcalls_GET_request/src/hello_http_rust_backend/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+                //         5.718, <-- high
+                //         5.714, <-- open
+                //         5.714, <-- close
+                //         243.5678 <-- volume
+                //     ],
+
+            //Return the body as a string and end the method
+            str_body
+        }
+        Err((r, m)) => {
```
Comment (adamspofford-dfinity) [inline rust/https_outcalls_GET_request/src/hello_http_rust_backend/src/lib.rs:81]: The motoko code doesn't try/catch, so this is a trap in Motoko but a successful return in Rust. This should probably be changed to an explicit trap, or just unwrap the http_request call.

File: rust/https_outcalls_GET_request/src/hello_http_rust_backend/src/lib.rs (hunk 3 above)
Comment (dprats) [reply rust/https_outcalls_GET_request/src/hello_http_rust_backend/src/lib.rs:16]: done

**Pull request merged**